### ISSUE TITLE
Add an alias to `InvalidPackageDeclaration` rule and fix `PackageNaming` rule alias

### ISSUE THREAD

**Issue by osipxd** [detekt/detekt#5540]
## Expected Behavior of the rule

- `PackageNaming` - should have an alias `PackageName` instead of `PackageDirectoryMismatch`
- `InvalidPackageDeclaration` - should have an alias `PackageDirectoryMismatch`

If this proposal is OK, I can implement it myself.

**Comment by schalkms** [comment]
Could you please elaborate why this alias should change for all users and not just for users who like this alias proposal?

**Comment by osipxd** [comment]
Ah, sorry 😅
IntelliJ IDEA provides similar inspections:
- `PackageName` - reports package names that do not follow the naming conventions
- `PackageDirectoryMismatch` - reports package directives that do not match the location of the file

It would be natural to have these aliases for detekt rules.

**Comment by cortinico** [comment]
> * should have an alias `PackageName` instead of `PackageDirectoryMismatch`

I would rather have 2 aliases rather than changing one as it's a breaking change.

>  IntelliJ IDEA provides similar inspections:

Feels like a bit of a weak argument to add this.

**Comment by osipxd** [comment]
> I would rather have 2 aliases rather than changing one as it's a breaking change.

👍 I agree

> Feels like a bit of a weak argument to add this.

Without alias, we will need to suppress both, IDEA inspection and detekt rule in case we have package naming violation:
```kotlin
@file:Suppress("PackageNaming", "PackageName")
```

Some of detekt rules already have aliases corresponding to IDEA inspections. Some of these rules:

| detekt rule | IDEA inspection |
|------------|-----------------|
| `ClassNaming` | `ClassName` |
| `FunctionNaming` | `FunctionName` |
| `Deprecation` | `DEPRECATION` |
| `UnsafeCast` | `UNCHECKED_CAST` |

**Comment by BraisGabin** [comment]
Thanks for the explanation. It have sense to me to add those aliases.

**Comment by cortinico** [comment]
Thanks for providing the reference table. I think we should add the alias you suggested + we should add comments on the rules you mentioned referencing the IDEA inspections.

**Comment by osipxd** [comment]
> \+ we should add comments on the rules you mentioned referencing the IDEA inspections.

I've updated the table to include all rules having aliases matching to IDEA inspections.

**Comment by BraisGabin** [comment]
Yes
